src/libraries/Common/src/System/Net/Security/Unix/SafeDeleteNegoContext.cs

```diff
@@ -59,6 +63,9 @@ public SafeDeleteNegoContext(SafeFreeNegoCredentials credential, string targetNa
                 Dispose();
                 throw;
             }
+            _credential = credential;
+            bool ignore = false;
+            _credential.DangerousAddRef(ref ignore);
```

Comment:
Where is DangerousRelease called?

Reply:
I used `_credential.Dispose` but that was probably incorrect. Since we don't own the credentials here, we should not Dispose. I modified the code to drop the reference count instead.

Reply:
Dispose and DangerousRelease are different.  In fact, if there's an outstanding reference count on the instance, Dispose will be a nop until that ref count drops to 0.

Reply:
yes, I realized that after the comment. I was lured by existing code in Dispose without realizing the pattern was different ;(